### PR TITLE
#164047070  user can favourite an article

### PR DESCRIPTION
**Pull request by Kibetchirchir**
#### Description
___
An authenticated user can favourite an article. They can also unfavorite an article dislike. 

#### Type of Change
___

- [x] New Feature

#### How has this been tested
___
- [x] Unit tests
 - [x]  End to end testing

#### Checklist
___
- [x] Tests are included to show/proof the feature works.

#### Pivotal Tracker
___
[164047070](https://www.pivotaltracker.com/story/show/164047070)

### CODE REVIEW

File: authors/apps/article/tests/test_favorite_article.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                         HTTP_AUTHORIZATION='Token ' + self.token,
+                         format='json')
+
+        response_article = self.client.delete('/api/articles/' + slug + '/favorite',
+                                              HTTP_AUTHORIZATION='Token ' + self.token,
+                                              format='json')
+        article = json.loads(response_article.content)
+
+        self.assertIn(
+            'you have successfull deleted', str(article))
```
Comment (JoanNgatia) [inline authors/apps/article/tests/test_favorite_article.py]: Nitpick fix: `you have successfully deleted`

File: authors/apps/article/views.py [hunk 2]
```diff
@@ -56,52 +54,10 @@ def get(self, request):
         Retrieve all articles
         """
         article = Article.objects.all()
-        serializer = GetArticleSerializer(article, many=True)
+        serializer = ArticleSerializer(article, many=True, context={'request': self.request})
         return Response(serializer.data)
 
 
-    def destroy(self, request, slug):
```
Comment (JoanNgatia) [inline authors/apps/article/views.py]: Also, @Kibetchirchir why are we removing these methods? Are they catered for in another class?

**Review by JoanNgatia — CHANGES_REQUESTED** [review]

File: authors/apps/article/views.py (hunk 2 above)
Comment (Kibetchirchir) [reply authors/apps/article/views.py]: The destroy and put method are not being called. I refactored the code from generics to API view and forgot to remove the unused methods

File: authors/apps/article/tests/test_favorite_article.py (hunk 1 above)
Comment (Kibetchirchir) [reply authors/apps/article/tests/test_favorite_article.py]: Thank you

File: authors/apps/article/views.py (hunk 2 above)
Comment (JoanNgatia) [reply authors/apps/article/views.py]: okay, Thanks

**Pull request merged**